bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -0,0 +1,214 @@
+                                    f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = np.concatenate([all_trig_ids[ifo],
+                              f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = np.concatenate([all_trig_times[ifo],
+                                 -1*np.ones_like(f_in['foreground/fap'][:],
+                                                    dtype=np.uint32)])
+            all_trig_ids[ifo] = np.concatenate([all_trig_ids[ifo],
+                                 -1*np.ones_like(f_in['foreground/fap'][:],
+                                                    dtype=np.uint32)])
+
+for ifo in all_ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+n_triggers = f['foreground/ifar'].size
+logging.info('{} triggers'.format(n_triggers))
+
+# all_times is a tuple of trigger time arrays
+all_times = (f['foreground/%s/time' % ifo][:] for ifo in all_ifos)
+
+# Currently only clustering zerolag, i.e. foreground, so set all timeslide_ids
+# to zero
+cidx = pycbc.events.cluster_coincs_multiifo(f['foreground/stat'][:], all_times,
+                                            np.zeros(n_triggers), 0,
+                                            args.cluster_window, argmax=np.argmax)
+
+
+def filter_dataset(h5file, name, idx):
+    # Dataset needs to be deleted and remade as it is a different size
+    filtered_dset = h5file[name][:][idx]
+    del h5file[name]
+    h5file[name] = filtered_dset
+
+# Downsample the foreground columns to only the loudest ifar between the
+# multiple files
+for key in f['foreground'].keys():
+    if key not in all_ifos:
+        filter_dataset(f, 'foreground/%s' % key, cidx)
+    else:  # key is an ifo
+        for k in f['foreground/%s' % key].keys():
+            filter_dataset(f, 'foreground/{}/{}'.format(key, k), cidx)
+
+n_triggers = f['foreground/ifar'].size
+
+times_tuple = (f['foreground/{}/time'.format(ifo)] for ifo in all_ifos)
+test_times = np.array([pycbc.events.mean_if_greater_than_zero(tc)[0]
+                          for tc in zip(*times_tuple)])
+
+is_in_combo_time = {}
+for key in f['segments']:
+    is_in_combo_time[key] = np.zeros(n_triggers)
+    if key.startswith('foreground') or key.startswith('background'):
+        continue
+    end_times = np.array(f['segments/%s/end' % key][:])
+    start_times = np.array(f['segments/%s/start' % key][:])
+    idx_within_segment = pycbc.events.indices_within_times(test_times,
+                                                           start_times,
+                                                           end_times)
+    is_in_combo_time[key][idx_within_segment] = np.ones_like(idx_within_segment)
+del idx_within_segment
+
+available_combos =[' '.join(sorted([key for key in is_in_combo_time if is_in_combo_time[key][i]])).encode('utf8') for i in np.arange(n_triggers)]
```

Comment:
split this up into (at least) 2 lines, may also be clearer to do an explicit iteration over the n_triggers over rather than a list comprehension. 
also, add a comment saying what 'available combos' is .. it appears to be an enormous string?